stdlib/public/SwiftShims/RefCount.h

```diff
@@ -238,14 +238,11 @@ struct RefCountBitOffsets;
 // 32-bit out of line
 template <>
 struct RefCountBitOffsets<8> {
-  // We reserve 1 bit (which we likely be using in future) to make the
-  // unowned field 31 bit. The reason is that unowned overflow checking does
-  // not work with 32 bit in the current implementation.
-  static const size_t ReservedShift = 0;
-  static const size_t ReservedBitCount = 1;
-  static const uint32_t ReservedMask = maskForField(Reserved);
```

Comment:
This change seems wrong to me. The ref count representation is basically an enum of "fast" and "slow" representations, and the "is immortal" bit is only set in the slow representation. So you haven't actually removed the reserved bit here.

Reply:
Hmm. I'm viewing it as a 2 bit discriminator:
00: fast
01: side table
10: unused
11: immortal

I can see that we have "half a bit" left as milseman would put it, but I don't see how we have the whole bit.

Reply:
(Michael G and I were discussing maybe using the remaining bit pattern to indicate thread-local)

Reply:
"Half a bit" sounds fine. I think I'd just prefer if there were separate sets of constants for each representation, rather than trying to pretend it's all one representation. The (64-bit) "slow" representation has way more free bits than the "fast" representation.

(By the way, thread-local doesn't sound like a good choice to me because we want to keep the fast path fast, which means only having to check one bit. Ideally it'd be used for something that doesn't affect normal retain/release but which is still useful to track without allocating a side table.)

Reply:
That's fair. I'll file something against myself to clean this up (…and maybe document why the compare and swap usage is correct at the same time since it's now tripped up at least 3 people)